Algebra.java

```diff
@@ -6,62 +6,182 @@
+   		//System.out.println(sqrt(0));
+		//System.out.println(sqrt(263169));
    		System.out.println(sqrt(76123));
+		System.out.println(sqrt(9));
+		System.out.println(sqrt(81));
 	}  
 
 	// Returns x1 + x2
 	public static int plus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if(x2 < 0) {
+			for(int i = 0; i < Math.abs(x2); i++) {
+				x1--;
+			}
+		} else {
+			for(int i = 0; i < x2; i++) {
+				x1++;
+			}
+		}
+		return x1;
 	}
 
 	// Returns x1 - x2
 	public static int minus(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
+		if(x2 < 0) {
+			for(int i = 0; i < Math.abs(x2); i++) {
+				x1++;
+			}
+		} else {
+			for(int i = 0; i < x2; i++) {
+				x1--;
+			}
+		}
+		return x1;
 	}
 
 	// Returns x1 * x2
 	public static int times(int x1, int x2) {
-		// Replace the following statement with your code
+		int mult = x1;
+		if(x1 == 1)
+		return x2;
+		else if(x2 == 1)
+		return x1;
+		else if(x1 == 0 || x2 == 0)
 		return 0;
+		else if(x1 == -1 && x2 == -1)
+		return 1;
+		else if(x1 == -1 && x2 == 1)
+		return -1;
+		else if(x1 == 1 && x2 == -1)
+		return -1;
+		else if(x1 == 1 && x2 == 1)
+		return 1;
+		else if(x1 < 0 && x2 < 0) {
+			for(int i = 1; i < Math.abs(x2); i++) {
+				mult = plus(mult, x1);
+			}
+			mult = Math.abs(mult);
+		} else if(x1 < 0 && x2 > 0) {
+			for(int i = 1; i < Math.abs(x2); i++) {
+				mult = plus(mult, x1);
+			}
+		} else if(x1 > 0 && x2 < 0) {
+			for(int i = 1; i < Math.abs(x2); i++) {
+				mult = plus(mult, x1);
+			}
+			mult = minus(0, mult);
+		} else {
+			for(int i = 1; i < Math.abs(x2); i++) {
+				mult = plus(mult, x1);
+			}
+		}
+		return mult;
 	}
 
 	// Returns x^n (for n >= 0)
 	public static int pow(int x, int n) {
-		// Replace the following statement with your code
-		return 0;
+		if(n == 0)
+		return 1;
+		if(n == 1)
+		return x;
+		int pow = x;
+		for(int i = 1; i < n; i++) {
+			pow = times(pow, x);
+		}
+		return pow;
 	}
 
 	// Returns the integer part of x1 / x2 
-	public static int div(int x1, int x2) {
-		// Replace the following statement with your code
+	public static int div(int x1, int x2) { 
+		if(x1 == 0)
 		return 0;
+		if(x2 == 1)
+		return x1;
+		int q = x1;
+		int count = 0;
+		if(x1 < 0 && x2 < 0){
+			x2 = Math.abs(x2);
+			x1 = Math.abs(x1);
+			q = x1;
+			while(q >= x2) {
+				q = minus(q, x2);
+				count++;
+			}
+			return count;
+		} else if((x1 < 0 && x2 > 0) || (x1 > 0 && x2 < 0)) { 
+			x2 = Math.abs(x2);
+			x1 = Math.abs(x1);
+			q = x1;
+			while(q >= x2) {
+				q = minus(q, x2);
+				count++;
+			}
+			return times(count, -1);
+		} else {
+			while(q >= x2) {
+				q = minus(q, x2);
+				count++;
+			}
+		}
+	    return count;
 	}
+		
 
 	// Returns x1 % x2
 	public static int mod(int x1, int x2) {
-		// Replace the following statement with your code
-		return 0;
-	}	
+        int q = x1 / x2;
+        int r = x1 - (q * x2);
+        if (r < 0) {
+            r += Math.abs(x2);
```

Comment:
Math.abs(x2) is Math operation.. You are asked to work without Math...
This is violation of Algebra guidelines -2.